com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/DebugUtility.java

```diff
@@ -242,4 +257,57 @@ public static void resumeThread(ThreadReference thread) {
             // the resume operation to this thread is meanness.
         }
     }
+
+    /**
+     * Encode an string array to a string as the follows.
+     *
+     * <p>source argument:
+     * <pre>["path=C:\\ProgramFiles\\java\\bin", "JAVA_HOME=C:\\ProgramFiles\\java"]</pre>
+     *
+     * <p>after encoded:
+     * <pre>"29\npath=C:\\ProgramFiles\\java\\bin30\nJAVA_HOME=C:\\ProgramFiles\\java"</pre>
+     *
+     * @param argument the string array arguments
+     * @return the encoded string
+     */
+    public static String encodeArrayArgument(String[] argument) {
```

Comment:
We already referenced the apache commons lib. We can still keep the function but please use  org.apache.commons.lang3.StringUtils.join as implementation

Reply:
My implementation is not to join the array but use the special rule to encode the string array. Can you explain why need to use `StringUtils.join` as implementation?

Reply:
We can pair two calls to achieve what we want: StringUtils.join(args, '\n') and String.Utils.split(str, '\n')

Reply:
if the environment variable may contain "\n" by itself, the join/split cannot handle this case.

Reply:
Use URLEncoder/Decoder to encode/decode string array.